src/test/java/org/springframework/data/redis/connection/lettuce/LettuceConnectionFactoryUnitTests.java

```diff
@@ -109,6 +111,28 @@ public void timeoutShouldBeSetCorrectlyOnClusterClient() {
 		}
 	}
 
+	@Test // DATAREDIS-930
+	public void portShouldBeReturnedProperlyBasedOnConfiguration() {
+
+		RedisConfiguration redisConfiguration = new RedisStandaloneConfiguration("localhost", 16379);
+
+		LettuceConnectionFactory connectionFactory = new LettuceConnectionFactory(redisConfiguration,
+				LettuceClientConfiguration.defaultConfiguration());
+
+		assertThat(connectionFactory.getPort(), is(16379));
+	}
+
+	@Test // DATAREDIS-930
+	public void hostNameShouldBeReturnedProperlyBasedOnConfiguration() {
+
+		RedisConfiguration redisConfiguration = new RedisStandaloneConfiguration("external");
+
+		LettuceConnectionFactory connectionFactory = new LettuceConnectionFactory(redisConfiguration,
+				LettuceClientConfiguration.defaultConfiguration());
+
+		assertThat(connectionFactory.getHostName(), is("external"));
+	}
```

Comment:
We should also add a test for a custom `RedisConfiguration` that implements `WithHostAndPort`.

Reply:
I will add this test :)

Reply:
Done, I have created a static nested class for the custom `RedisConfiguration` let me know if that is ok or do you want to handle this in a different way, thanks